doc/design/diagrams/request_response_options.puml

```diff
@@ -0,0 +1,34 @@
+@startuml
+skinparam backgroundColor #EEEBDC
+
+class ClientOptions {
+  responseQueueCapacity: uint64_t
+  nodeName: NodeName_t
+  connectOnCreate: bool
+  responseQueueFullPolicy: ResponseQueueFullPolicy
+  serverTooSlowPolicy: ServerTooSlowPolicy
+}
+
+class ServerOptions {
+  requestQueueCapacity: uint64_t
+  nodeName: NodeName_t
+  offerOnCreate: bool
+  clientTooSlowPolicy: ClientTooSlowPolicy
+}
+
+enum ResponseQueueFullPolicy {
+    BLOCK_SERVER,
+    DISCARD_OLDEST_DATA
+}
+
+enum ClientTooSlowPolicy {
+    WAIT_FOR_CLIENT,
+    DISCARD_OLDEST_DATA
+}
+
+enum ServerTooSlowPolicy {
+    WAIT_FOR_SERVER,
+    DISCARD_OLDEST_DATA
```

Comment:
For publish-subscribe we already have `QueueFullPolicy` and `SubscriberTooSlowPolicy`. Since they are also used in the `ChunkSender` it's cumbersome to always do the conversion. I would suggest to combine all this enums into
```
enum QueueFullPolicy {
    BLOCK_PARTICIPANT, // or BLOCK_WRITER
    DISCARD_OLDEST_DATA
}
```
and
```
enum ParticipantTooSlowPolicy { // or ReaderTooSlowPolicy
    WAIT_FOR_PARTICIPANT, // or WAIT_FOR_READER
    DISCARD_OLDEST_DATA
}
```
The corresponding member would still be called `subscriberTooSlowPolicy`. What do you think?

Reply:
Okay, it's now
```cpp
enum QueueFullPolicy {
    BLOCK_PRODUCER,
    DISCARD_OLDEST_DATA
};

enum ConsumerTooSlowPolicy {
    WAIT_FOR_CONSUMER,
    DISCARD_OLDEST_DATA
};
```